app/serializers/topic_view_serializer.rb

```diff
@@ -213,6 +214,10 @@ def unpinned
     PinnedCheck.unpinned?(object.topic, object.topic_user)
   end
 
+  def ignored_usernames
+    IgnoredUser.where(user: topic.user).joins(:ignored_user).pluck(:username)
```

Comment:
This is N+1 but I have checked against when is `TopicViewSerializer` called, and it seems to be that it called in the `TopicsController` primarily, which loads once.

I would love some thoughts if this is okay or we want to avoid it completely, and if we want to avoid what other possible ways to achieve this?

Reply:
We definitely want to avoid these time 💣 if possible. Usually, to avoid N+1 we preload records. Maybe look at how we preload custom fields?